build/print.go

```diff
@@ -330,7 +330,9 @@ func isDifferentLines(p1, p2 *Position) bool {
 // binds tighter than a smaller number. All binary operators bind
 // left-to-right.
 const (
-	precLow = iota
+	precType = iota
+	precDefault
```

Comment:
This is a slight abuse of operator precedence, but I did it this way to avoid having to make extensive changes to the AST-walking code. It works because `precType` and `precDefault` always have lower precedence than `precLow` (which is the lowest precedence that the upstream code uses), so there's no risk of them accidentally being used while a complex expression is printed that has nothing to do with function parameter types or defaults - see line 627 for an idea of how the `prec*` constants are used, and how defining them another way might lead to unexpected output.